transport/transport.go

```diff
@@ -303,6 +303,24 @@ func StreamFromContext(ctx context.Context) (s *Stream, ok bool) {
 	return
 }
 
+// Peer connection info
+type Peer struct {
+	Addr net.Addr
+}
+
+type peerCtxKey struct{}
+
+// Save peer info in a new context and return it
+func newContextWithPeer(ctx context.Context, peer Peer) context.Context {
+	return context.WithValue(ctx, peerCtxKey{}, peer)
+}
+
+// Retrieve peer info from context
```

Comment:
starts the comment with "PeerFromContext". So probably

// PeerFromContext retrieves the peer information from ctx.